tensorflow-core/tensorflow-core-api/src/bazel/op_generator/op_generator.cc

```diff
@@ -246,6 +246,10 @@ void RenderFactoryMethods(const OpSpec& op, const Type& op_class,
       writer->EndLine();
     }
   }
+  // Add Device String
+  writer->Append("opBuilder.setDevice(scope.makeDeviceString());");
+  writer->EndLine();
+
```

Comment:
Hey @zaleslaw , I have another change to propose you that I just thought of. I think if we merge together `scope.applyControlDependencies` and this new step for adding the device string to the `opBuilder` would make the whole process of building an op simpler, especially with non-generated ones.

So what about that:
1. We rename `scope.applyControlDependencies(opBuilder)` to `scope.apply(opBuilder)`
2. In `Scope.apply`, we apply both control dependencies, the device string (and eventually more data if needed)

By doing so, we also won't need `scope.makeDeviceString()` anymore. Let me know what you think